gtdynamics/kinematics/KinematicsSlice.cpp

```diff
@@ -98,38 +120,94 @@ EqualityConstraints Kinematics::pointGoalConstraints<Slice>(
   return constraints;
 }
 
+template <>
+NonlinearFactorGraph Kinematics::poseGoalObjectives<Slice>(
+    const Slice& slice, const Robot& robot,
+    const gtsam::Values& goal_poses) const {
+  gtsam::NonlinearFactorGraph graph;
+
+  // Add priors on link poses with desired poses from argument
+  for (auto&& link : robot.links()) {
```

Comment:
The problem I see is that goal_poses might carry the pose objectives of a whole interval, so there would be no way to distinguish which poses are for the current time-frame. This would mean that we would be adding pose priors on other slices if we were to use this example code.

However, you are right that the robot doesn't really do much. Since I will change these Values to some other container (contactPoses or similar), I'll keep this in mind.